tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -68,6 +72,29 @@ def _initialize_lisa_logger():
 _initialize_lisa_logger()
 
 
+#
+# Helper to change the current thread name temporarily
+#
+@contextlib.contextmanager
+def _set_thread_name(name: str):
+    initial_name = current_thread().name
+    current_thread().name = name
+    try:
+        yield
+    finally:
+        current_thread().name = initial_name
+
+
+#
+# Possible values for the collect_logs parameter
+#
+class CollectLogs(Enum):
+    Always = 'always'   # Always collect logs
+    Failed = 'failed'   # Collect logs only on test failures
+    No = 'no'           # Never collect logs
+
```

Comment:
These enum values match the values for LISA's 'keep_environment' runbook variable (which is used to prevent LISA from deleting the test VMs at the end of the runbook)